tensorflow/core/kernels/data/batch_dataset_op.cc

```diff
@@ -12,298 +12,303 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+            PartialTensorShape({-1}).Concatenate(input_shape));
+      }
     }
   }
 
-  void MakeDataset(OpKernelContext* ctx, DatasetBase* input,
-                   DatasetBase** output) override {
-    int64 batch_size = 0;
-    OP_REQUIRES_OK(ctx,
-                   ParseScalarArgument<int64>(ctx, "batch_size", &batch_size));
-    OP_REQUIRES(
-        ctx, batch_size > 0,
-        errors::InvalidArgument("Batch size must be greater than zero."));
-
-    bool drop_remainder = false;
-    if (op_version_ > 1) {
-      OP_REQUIRES_OK(ctx, ParseScalarArgument<bool>(ctx, "drop_remainder",
-                                                    &drop_remainder));
-    }
+  ~Dataset() override { input_->Unref(); }
 
-    *output =
-        new Dataset(ctx, batch_size, drop_remainder, parallel_copy_, input);
+  std::unique_ptr<IteratorBase> MakeIteratorInternal(
+      const string& prefix) const override {
+    return absl::make_unique<Iterator>(Iterator::Params{
+        this, name_utils::IteratorPrefix(kDatasetType, prefix)});
   }
 
- private:
-  class Dataset : public DatasetBase {
-   public:
-    Dataset(OpKernelContext* ctx, int64 batch_size, bool drop_remainder,
-            bool parallel_copy, const DatasetBase* input)
-        : DatasetBase(DatasetContext(ctx)),
-          batch_size_(batch_size),
-          drop_remainder_(drop_remainder),
-          parallel_copy_(parallel_copy),
-          input_(input) {
-      input_->Ref();
-
-      // NOTE(mrry): Currently we implement "batch up to" semantics. If
-      // we could tell statically that the input dataset is infinite,
-      // then we could always report `batch_size` as the 0th dimension.
-      const auto& input_shapes = input_->output_shapes();
-      output_shapes_.reserve(input_shapes.size());
-      for (const auto& input_shape : input_shapes) {
-        if (drop_remainder_) {
-          output_shapes_.emplace_back(
-              PartialTensorShape({batch_size_}).Concatenate(input_shape));
-        } else {
-          output_shapes_.emplace_back(
-              PartialTensorShape({-1}).Concatenate(input_shape));
-        }
-      }
-    }
-
-    ~Dataset() override { input_->Unref(); }
+  const DataTypeVector& output_dtypes() const override {
+    return input_->output_dtypes();
+  }
 
-    std::unique_ptr<IteratorBase> MakeIteratorInternal(
-        const string& prefix) const override {
-      return absl::make_unique<Iterator>(
-          Iterator::Params{this, strings::StrCat(prefix, "::Batch")});
-    }
+  const std::vector<PartialTensorShape>& output_shapes() const override {
+    return output_shapes_;
+  }
 
-    const DataTypeVector& output_dtypes() const override {
-      return input_->output_dtypes();
-    }
+  string DebugString() const override {
+    name_utils::DatasetDebugStringParams params;
+    params.op_version = op_version_;
+    params.set_args(batch_size_);
+    return name_utils::DatasetDebugString(kDatasetType, params);
```

Comment:
Just want you to notice that `DebugString()` will return a different string than before. For `BatchDatasetV2`, it returns `BatchDatasetV2Op(batch_size)::Dataset`; For `BatchDataset`, it returns `BatchDatasetOp(batch_size)::Dataset`. Before, it always returns `BatchDatasetOp(batch_size)::Dataset`.